digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -1499,6 +1510,8 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
 
       val (generatedLinks, adjustedLinks) = (recalculated ++ assignedTerminatedRoadwayNumbers).partition(_.id == NewIdValue)
 
+      roadwayChangesDAO.clearRoadChangeTable(projectId)
```

Comment:
I'm not sure if I understood the need for this, and why it was not needed before?

Reply:
**CHANGE_TABLE** data, for a certain project, is produced from scratch every single time an user changes anything within project links. Because, in this process we need to revert all the project links that were split we will need to delete some lines from the table **PROJECT_LINK**.

Because change_table and project_link are connected by **FK** reference, the best way to do it, is to clear the change table first:

`roadwayChangesDAO.clearRoadChangeTable(projectId)
projectLinkDAO.removeProjectLinksById(splitLinks.map(_.id).toSet)`

Reply:
Ok, got it.